src/code/z_fireobj.c

```diff
@@ -1,27 +1,244 @@
+    FireObjLightParams* temp_v1 = &sFireObjLightParams[fire->unk27];
+
+    if (fire->state != 3) {
+        Vec3s vec;
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        func_8012C2DC(globalCtx->state.gfxCtx);
+        gSPSegment(POLY_XLU_DISP++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0, (fire->unk26 * -0x14) & 0x1FF,
+                                    0x20, 0x80));
+
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, temp_v1->unk4, temp_v1->unk0, temp_v1->unk1, temp_v1->unk2, temp_v1->unk3);
+
+        gDPSetEnvColor(POLY_XLU_DISP++, temp_v1->unk5, temp_v1->unk6, temp_v1->unk7, 0);
+
+        vec.x = 0;
+        vec.y = func_800DFCDC(globalCtx->cameraPtrs[globalCtx->activeCamera]) + 0x8000;
+        vec.z = 0;
+        Matrix_SetStateRotationAndTranslation(fire->position.x, fire->position.y, fire->position.z, &vec);
+        Matrix_Scale(fire->xScale, fire->yScale, 1.0f, 1);
+
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, D_0407D590);
+        CLOSE_DISPS(globalCtx->state.gfxCtx);
+    }
+}
+
+#ifdef NON_EQUIVALENT
+// Accesses data incorrectly
+void FireObj_InitLight(GlobalContext* globalCtx, FireObjLight* light, u8* param_3, Vec3f* pos) {
+    FireObjLightParams* objectParams;
+
+    objectParams = &sFireObjLightParams[*param_3];
+
+    Lights_PointGlowSetInfo(&light->lightInfo, pos->x, pos->y, pos->z, objectParams->r, objectParams->g,
+                            objectParams->b, objectParams->radius);
+    light->light = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, (LightInfo*)&light->lightInfo);
+    light->unk12 = *param_3;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_fireobj/FireObj_InitLight.s")
+#endif
+
+void FireObj_DestroyLight(GlobalContext* globalCtx, FireObjLight* light) {
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, light->light);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fireobj/FireObj_FiniLight.s")
+#ifdef NON_EQUIVALENT
+// Accesses data incorrectly
+void FireObj_UpdateLight(GlobalContext* globalCtx, FireObjLight* light, FireObj* fire) {
+    FireObjLightParams* lightParams = &sFireObjLightParams[light->unk12];
+    s16 radius;
 
+    if (fire->state == 3) {
+        Lights_PointSetColorAndRadius((LightInfo*)&light->lightInfo, 0U, 0U, 0U, (s16)-1);
+    } else {
+        radius = ((fire->yScale * 140.0f * fire->sizeInv) + 60.0f);
+
+        Lights_PointGlowSetInfo((LightInfo*)&light->lightInfo, fire->position.x,
+                                (fire->position.y + (fire->yScale * 6500.0f)), fire->position.z,
+                                ((s32)(Rand_ZeroOne() * lightParams->unk5) + lightParams->unk2),
+                                ((s32)(Rand_ZeroOne() * lightParams->unk6) + lightParams->unk3),
+                                ((s32)(Rand_ZeroOne() * lightParams->unk7) + lightParams->unk4), radius);
```

Comment:
It should be pretty clear what most or all of the `FireObjLightParams` fields are for now.